src/lib/Movement/moveTo.ts

```diff
@@ -92,6 +92,77 @@ export const moveTo = (
     clearCachedPath(creep, cache);
   }
 
+  // Dynamic choose weight for roads, plains and swamps depending on body.
+  // Skip if power creep or any costs has been manually set.
+  if (DEBUG) logCpu('getting default plain and swamp costs');
+  const manuallyDefinedCosts = opts?.roadCost || opts?.plainCost || opts?.swampCost;
```

Comment:
These could be set to 0, which would be incorrect for other reasons, but to avoid unexpected issues this should explicitly check for undefined:

```
const manuallyDefinedCosts = (opts?.roadCost !== undefined || opts?.plainCost !== undefined || opts?.swampCost !== undefined);
```